src/app/qgisapp.cpp

```diff
@@ -13409,3 +13419,51 @@ QgsFeature QgisApp::duplicateFeatureDigitized( QgsMapLayer *mlayer, const QgsFea
   return QgsFeature();
 }
 
+
+void QgisApp::populateProjectStorageMenu( QMenu *menu, bool saving )
+{
+  menu->clear();
+  for ( QgsProjectStorage *storage : QgsApplication::projectStorageRegistry()->projectStorages() )
```

Comment:
This will detach (I suggest using qt creator 4.6 and enable clazy warnings)

Reply:
thanks for the tip... looks like I should upgrade!